tests/ibc-hooks/xcs_cw20_test.go

```diff
@@ -0,0 +1,179 @@
+package ibc_hooks_test
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v4/modules/apps/transfer/types"
+	ibctesting "github.com/cosmos/ibc-go/v4/testing"
+	"github.com/osmosis-labs/osmosis/v15/tests/osmosisibctesting"
+)
+
+// Instantiate the cw20 and cw20-ics20 contract
+func (suite *HooksTestSuite) SetupCW20(chainName Chain) (sdk.AccAddress, sdk.AccAddress) {
+	// Instantiate the cw20 contract on chainB
+	chain := suite.GetChain(chainName)
+	cw20CodeId := chain.StoreContractCodeDirect(&suite.Suite, "./bytecode/cw20_base.wasm")
+	// create contract with initial balances
+	cw20Addr := chain.InstantiateContract(&suite.Suite,
+		fmt.Sprintf(`{"name": "CW20-token0", "symbol": "cwtoken", "decimals": 6, "initial_balances": [{"address": "%s", "amount": "1000000000000000000000000"}]}`,
+			chain.SenderAccount.GetAddress().String()), cw20CodeId)
+	// Store the cw20-ics20 contract on chainB
+	cw20ics20CodeId := chain.StoreContractCodeDirect(&suite.Suite, "./bytecode/cw20_ics20.wasm")
+	// Instantiate the cw20-ics20 contract on chainB
+	cw20ics20Addr := chain.InstantiateContract(&suite.Suite,
+		fmt.Sprintf(`{"default_timeout": 1200, "gov_contract": "%s", "allowlist": [], "default_gas_limit": 200000}`,
+			chain.SenderAccount.GetAddress().String()), cw20ics20CodeId)
+	return cw20Addr, cw20ics20Addr
+}
+
+// Function to easily transfer the created cw20 tokens to chainA
+func (suite *HooksTestSuite) TransferCW20Tokens(path *ibctesting.Path, cw20Addr, cw20ics20Addr, receiver sdk.AccAddress, amount, memo string) (*sdk.Result, []byte) {
+	chainB := suite.GetChain(ChainB)
+
+	osmosisApp := chainB.GetOsmosisApp()
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(osmosisApp.WasmKeeper)
+
+	if len(memo) == 0 {
+		memo = "\"\""
+	}
```

Comment:
for my knowledge, why is the memo required?

Reply:
because the empty string would Sprintf to nothing and then the json would say `"memo": }` which is invalid. We could remove the memo key altogether, but thought this was cleaner than two srpintfs